src/components/Box/Box.styled.ts

```diff
@@ -0,0 +1,18 @@
+import styled from "styled-components";
+import { Size } from "./types";
+import { BoxProps } from "./Box";
+
+export const StyledBox = styled.div<BoxProps>`
+  position: relative;
+  display: ${(p) => p.display || "block"};
+  flex-direction: ${(p) => p.direction || "row"};
+
+  gap: ${(p) => (p.gap ? Size[p.gap] : 0)};
+  justify-content: ${(p) => p.justifyContent || "start"};
+  align-items: ${(p) => p.alignItems || "strech"};
+  align-self: ${(p) => p.alignSelf || "auto"};
+
+  flex-grow: ${(p) => p.grow || 0};
+  flex-shrink: ${(p) => p.shrink || 0};
```

Comment:
Not sure if 0 is used intentionally or to set the initial value if it is not then can use `initial` or `unset` for the default values